### PR TITLE
Project: Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by dipbazz**
In the project, we have created a form where user can be created, edited but not deleted.

The model contains a `User` model with fields `username`, `email`, and `password`, and these fields are validated to be present.

The controller has methods `#new`, `#edit`, `#update`, and `#create`.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,35 @@
 class UsersController < ApplicationController
   def new
+    @user = User.new
   end
 
   def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb]: - _Remove commented code_

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to new_user_path
```
Comment (lekegitrepo) [inline app/controllers/users_controller.rb]: - _This should be `redirect_to edit_user_path` if the user was saved not `new_user_path`_

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<%
+=begin%>
+ <form action="/users" method="post" accept-charset="UTF-8">
```
Comment (lekegitrepo) [inline app/views/users/new.html.erb]: - _This should use `users_path` helper for action_

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+<%= label_tag(:password, "password") %>
+<%= password_field_tag(:password) %>
+
+<%= submit_tag("submit") %>
+
+<% end %> 
+<%
+=end%>
+
+<%= form_for(User.new) do |f| %>
```
Comment (lekegitrepo) [inline app/views/users/new.html.erb]: - _This should use `@user` from the controller_

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @dipbazz and @sneeu-leeu :clap: :clap:
#### Well done Team, you have done a good job so far. The readme looks great. However, there are minor requirements you need to implement. Check my comments below :point_down:

#### Commented form:
- _Use the helper method for the form action and instance variable from the controller_

#### UsersController:
- Use the `edit_user_path` instead of the `new_user_path`

#### Commented Code:
- Remove any commented codes from your code

### Optional:
- _Use partial for the user's form and reuse it in the edit and new views_
- _Add validation to the form to display a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))_

#### _Reviewed by Temitayo Adeleke_

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,7 +1,34 @@
...
```
Comment (SafaErden) [inline app/controllers/users_controller.rb]: - [x] This should be redirect_to edit_user_path if the user was saved not new_user_path as the previous TSE mentioned. I think you forgot, please kindly fix it.

**Review by SafaErden — APPROVED** [review]
## Status: Approved 🥳🎉🎊

**Hi TEAM, Nice job implementing the required changes! The app looks amazing. I know that takes a lot of time and patience.**
##

**Take into account the following before merging your PR🧐:**

You missed implementing one required change mentioned by the previous TSE, Please fix it before merging your project. Check below for details.

##

Questions or comments? Reach me on ([Twitter](https://twitter.com/safaerden)) or ([LinkedIn](https://www.linkedin.com/in/safaerden/)) 🗨️

Happy coding_ 👨‍💻 

**Safa ERDEN**

**Pull request merged**